Create a Friendly URL for /redaccion.php

isabelccc opened the pull request:
# Description

add statement_editor.php under problem folder, add rewrite entry in nginx to redirect 

If you modify the **User Interface**, please include a screenshot or a video.
<img width="528" alt="Screenshot 2025-04-05 at 1 24 35 AM" src="https://github.com/user-attachments/assets/6f26b80f-7c05-4c8b-9be1-ee7e7c570b13" />

Fixes: #8146 



# Comments

Add any comments for the reviewers (e.g., indicating the beginning of the revision, something where the reviewer needs to pay special attention, etc.). If  there are no extra comments this section can be deleted.

# Checklist:

- [ ✅] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [ ] The tests were executed and all of them passed.
- [ ] If you are creating a feature, the new tests were added.
- [ ] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

pabo99 commented on frontend/database/dao_schema.sql:
```diff
@@ -972,6 +972,7 @@ CREATE TABLE `Schools` (
   UNIQUE KEY `name_country_id_state_id` (`name`,`country_id`,`state_id`),
   KEY `country_id` (`country_id`),
   KEY `state_id` (`country_id`,`state_id`),
+  KEY `idx_schools_name` (`name`),
```
This change doesn’t belong in this PR. Could it be reverted?
```suggestion
```

pabo99 commented on frontend/server/nginx.rewrites:
```diff
@@ -2,6 +2,8 @@ location /api/ {
 	rewrite ^/api/(.*)$ /api/ApiEntryPoint.php last;
 }
 
+
+
```
This is not needed.
```suggestion
```

pabo99 commented on frontend/server/nginx.rewrites:
```diff
@@ -70,6 +72,7 @@ rewrite ^/group/([a-zA-Z0-9_+:-]+)/edit/?$ /group/edit.php?group=$1 last;
 rewrite ^/group/([a-zA-Z0-9_+:-]+)/scoreboard/([a-zA-Z0-9_+-]+)/?$ /groupscoreboard.php?group=$1&scoreboard=$2 last;
 rewrite ^/group/([a-zA-Z0-9_+:-]+)/scoreboard/([a-zA-Z0-9_+-]+)/edit/?$ /group/scoreboardedit.php?group=$1&scoreboard=$2 last;
 rewrite ^/group/([a-zA-Z0-9_+:-]+)/stats/?$ /groupstats.php?group=$1 last;
+rewrite ^/redaccion.php$ /problems/statement_editor.php permanent;
```
Could you add a new entry point with a user-friendly URL?
This one could be: `/problem/statement/editor/`

pabo99 commented on frontend/server/src/DAO/Schools.php:
```diff
@@ -32,7 +32,7 @@ public static function findByName($name) {
                 Schools s
             WHERE
-                s.name LIKE CONCAT(\'%\', ?, \'%\')
+                s.name LIKE CONCAT(?, \'%\')
```
This change isn't needed in this PR either, so it should be reverted as well.
```suggestion
```

pabo99 commented on frontend/www/problems/statement_editor.php:
```diff
@@ -0,0 +1,15 @@
+require_once(dirname(__DIR__, 2) . '/server/bootstrap.php');
+
+\OmegaUp\UITools::render(
+    fn (\OmegaUp\Request $r) => [
+        'templateProperties' => [
+            'title' => new \OmegaUp\TranslationString(
+                'omegaupTitleStatementEditor',
+            ),
+            'payload' => [],
+        ],
+        'entrypoint' => 'problem_statement',
+    ]
```
PSALM might complain that the return type is not declared. Can you add the one used in the file `redaccion.php`?

isabelccc replied:
Ok , I have fixed all above suggestions

pabo99 reviewed (commented):
Can you run the following command after applying the changes?
```bash
./stuff/lint.sh fix -- frontend/www/problems/statement_editor.php
```
This will help fix some of the errors reported by GitHub.

pabo99 reviewed (approved):
Thank you!

isabelccc commented:
@pabo99 , Hi Juan Pablo, this is completed, pls check :) let know if there are other suggestions

pabo99 commented:
> @pabo99 , Hi Juan Pablo, this is completed, pls check :) let know if there are other suggestions

Hi, @isabelccc ! I approved it yesterday! 

Defer to @heduenas

pabo99 commented:
@isabelccc could you please update your branch?

heduenas commented:
@isabelccc some conflicts have arised, can you please resolve them?

isabelccc commented:
> @isabelccc some conflicts have arised, can you please resolve them?

ok no prob, I'll fix it up today

pabo99 commented:
Please run:
```bash
./stuff/lint.sh fix -- frontend/www/problems/statement_editor.php
```
outside the container in order to fix the linter errors

pabo99 commented on frontend/www/problems/statement_editor.php:
```diff
@@ -1,16 +1,15 @@
+                'omegaupTitleStatementEditor',
             ),
             'payload' => [],
-            'fullWidth' => true,
```
This line was added in another PR, could you please restore it?

isabelccc commented:
> Please run:
> 
> ```shell
> ./stuff/lint.sh fix -- frontend/www/problems/statement_editor.php
> ```
> 
> outside the container in order to fix the linter errors
@pabo99 
I found out i  FAIL frontend/www/js/omegaup/arena/ranking.test.ts (6.75 s), this test when i added 'fullWidth' => true,so i removed it early on (If i remove it, it passes without issues) , now this issue come back , shall we still keep 'fullWidth' => true," there ?

pabo99 commented:
@isabelccc Oh, I understand the reason why you made the change. Let me tell you that this test has been showing inconsistencies, which is causing the JavaScript check to fail. PR #8218 has already been created to fix it. Once it is approved and merged into the main branch, this issue shouldn't happen again.

isabelccc commented:
> @isabelccc Oh, I understand the reason why you made the change. Let me tell you that this test has been showing inconsistencies, which is causing the JavaScript check to fail. PR #8218 has already been created to fix it. Once it is approved and merged into the main branch, this issue shouldn't happen again.

@pabo99  ,OK thank you so much for the guidance
